src/pages/settings/Profile/ProfilePage.js

```diff
@@ -153,52 +125,72 @@ class ProfilePage extends Component {
+
+        this.setState({isAvatarChanged: true});
     }
 
     /**
      * Submit form to update personal details
+     * @param {Object} values
      */
-    updatePersonalDetails() {
-        if (!this.validateInputs()) {
-            return;
-        }
-
+    updatePersonalDetails(values) {
         // Check if the user has modified their avatar
-        if ((this.props.currentUserPersonalDetails.avatar !== this.state.avatar.uri) && this.state.isAvatarChanged) {
+        if ((this.props.currentUserPersonalDetails.avatar !== this.avatar.uri) && this.state.isAvatarChanged) {
             // If the user removed their profile photo, replace it accordingly with the default avatar
-            if (this.state.avatar.uri.includes('/images/avatars/avatar')) {
-                PersonalDetails.deleteAvatar(this.state.avatar.uri);
+            if (this.avatar.uri.includes('/images/avatars/avatar')) {
+                PersonalDetails.deleteAvatar(this.avatar.uri);
             } else {
-                PersonalDetails.setAvatar(this.state.avatar);
+                PersonalDetails.setAvatar(this.avatar);
             }
 
             // Reset the changed state
             this.setState({isAvatarChanged: false});
         }
 
         PersonalDetails.setPersonalDetails({
-            firstName: this.state.firstName.trim(),
-            lastName: this.state.lastName.trim(),
-            pronouns: this.state.pronouns.trim(),
+            firstName: values.firstName.trim(),
+            lastName: values.lastName.trim(),
+            pronouns: (this.state.hasSelfSelectedPronouns) ? values.selfSelectedPronoun.trim() : values.pronouns.trim(),
             timezone: {
-                automatic: this.state.isAutomaticTimezone,
-                selected: this.state.selectedTimezone,
+                automatic: values.isAutomaticTimezone,
+                selected: values.timezone,
             },
         }, true);
     }
 
-    validateInputs() {
-        const [hasFirstNameError, hasLastNameError, hasPronounError] = ValidationUtils.doesFailCharacterLimit(
-            50,
-            [this.state.firstName.trim(), this.state.lastName.trim(), this.state.pronouns.trim()],
+    /**
+     * @param {Object} values - An object containing the value of each inputID
+     * @returns {Object} - An object containing the errors for each inputID
+     */
+    validate(values) {
+        const errors = {};
+
+        const [hasFirstNameError, hasLastNameError, hasPronounError] = ValidationUtils.doesFailCharacterLimitAfterTrim(
+            CONST.FORM_CHARACTER_LIMIT,
+            [values.firstName, values.lastName, values.pronouns],
         );
+
+        const hasSelfSelectedPronouns = values.pronouns === CONST.PRONOUNS.SELF_SELECT;
+        this.pronouns = hasSelfSelectedPronouns ? '' : values.pronouns;
         this.setState({
-            hasFirstNameError,
-            hasLastNameError,
-            hasPronounError,
+            hasSelfSelectedPronouns,
```

Comment:
I'm not sure this is the best place to call this `setState` and update pronouns. Ideally the validate function should just validate the input. We call `validate` on change after the input is blurred and I think this might cause a lot of re-renders. I'm not sure that I have a good suggestion for where to call this instead.

Reply:
During my testing (and yours too), we found that Self selected pronoun wasn't behaving as it should. That's the reason this was moved to validate to fix this issue.

Reply:
Makes sense. I still don't think this is the best place to call that, but I recognize that <Form /> doesn't support that at the time. I created an [issue](https://github.com/Expensify/App/issues/10267) to add this feature to Form. I think for now this is ok and we would clean it up as part of that issue!